### ISSUE TITLE
Local queue service will not load

### ISSUE THREAD

**Issue by gallowasusc** [microsoft/AzureStorageExplorer#8392]
### Preflight Checklist

- [x] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [x] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [x] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.36.2

### Architecture

arm64

### Storage Explorer Build Number

20241029.2

### Platform

macOS

### OS Version

macOS 14.6.1 (23G93)

### Bug Description

Queue Service times out when attempting to connect to local azurite container instance.

### Steps to Reproduce

1. Create docker compose file with latest azurite instance (referenced below)
2. $> docker compose up
3. Open Storage Explorer
4. Select Emulator & Attached > Storage Accounts > docker emulator instance (or Emulator - Default Ports) > open tree for "queues" > timeout after 15 minutes.

### Actual Experience

The "queue" service does not successfully perform a GET request to the azurite instance to retrieve a list of queues available. It gets stuck loading. After 15 minutes, a timeout error is produced, referenced below:

```
Unable to retrieve child resources

ProducerError:{
  "name": "Node Fetch Error",
  "message": "Headers Timeout Error",
  "stack": "HeadersTimeoutError: Headers Timeout Error\n    at Ev.onParserTimeout [as callback] (/private/var/folders/j4/fsxxqtb925v8607vkld7fdhr0000gn/T/AppTranslocation/640BC08B-45C8-496D-A003-E3CC92B1C318/d/Microsoft Azure Storage Explorer.app/Contents/Resources/app/out/app/index.js:73:11116)\n    at Timeout.onTimeout [as _onTimeout] (/private/var/folders/j4/fsxxqtb925v8607vkld7fdhr0000gn/T/AppTranslocation/640BC08B-45C8-496D-A003-E3CC92B1C318/d/Microsoft Azure Storage Explorer.app/Contents/Resources/app/out/app/index.js:62:38070)\n    at listOnTimeout (node:internal/timers:581:17)\n    at process.processTimers (node:internal/timers:519:7)",
  "cause": {},
  "code": "UND_ERR_HEADERS_TIMEOUT"
}
```

Blob and Table services work as expected. I can create, read, update, and delete containers, tables, rows, etc. 

Other services, such as a custom web app, can CRUD from queue service hosted in container instance. Only the storage explorer times out attempting to connect. 

### Expected Experience

When selecting the "Queues" tree, the expectation is that a list of available queues will be populated. In my example provided in additional context, the expectation is that the storage explorer will determine that there are no queues available and provide me with the option to create a new queue.

### Additional Context

No settings changes to Azure Storage Explorer (I didn't even know it had a dark mode now lol)

Docker instance is version 4.36 (latest as of today)

Azurite instance is latest (3.33), determined by opening container -> terminal -> "$> azurite --version"

docker-compose.yaml:

```
  services:
    azurite-serve:
      image: mcr.microsoft.com/azure-storage/azurite
      container_name: azurite-internalapp
      ports:
        - "10000:10000"
        - "10001:10001"
        - "10002:10002"
      command: 'azurite --location /data --debug /data/azurite_debug.log --blobHost 0.0.0.0  --queueHost 0.0.0.0 --tableHost 0.0.0.0'
      volumes:
        - ./storage/azurite:/data
      networks:
        - internalapp
  networks:
    internalapp:
```

volume ./storage is empty and created when containers are spun up. Also fails when connecting to directory with previous local azure storage. 

Messages produced by the following:

`$> docker compose up`

```
[+] Running 1/0
 ✔ Container azurite-internalapp  Created                                                                                                  0.0s 
Attaching to azurite-internalapp
azurite-internalapp  | Azurite Blob service is starting at http://0.0.0.0:10000
azurite-internalapp  | Azurite Blob service is successfully listening at http://0.0.0.0:10000
azurite-internalapp  | Azurite Queue service is starting at http://0.0.0.0:10001
azurite-internalapp  | Azurite Queue service is successfully listening at http://0.0.0.0:10001
azurite-internalapp  | Azurite Table service is starting at http://0.0.0.0:10002
azurite-internalapp  | Azurite Table service is successfully listening at http://0.0.0.0:10002
```

After selecting tree for "azurite-internalapp", then opening tree for blob followed by table:
  
```
[+] Running 1/0
 ✔ Container azurite-internalapp  Created                                                                                                  0.0s 
Attaching to azurite-internalapp
azurite-internalapp  | Azurite Blob service is starting at http://0.0.0.0:10000
azurite-internalapp  | Azurite Blob service is successfully listening at http://0.0.0.0:10000
azurite-internalapp  | Azurite Queue service is starting at http://0.0.0.0:10001
azurite-internalapp  | Azurite Queue service is successfully listening at http://0.0.0.0:10001
azurite-internalapp  | Azurite Table service is starting at http://0.0.0.0:10002
azurite-internalapp  | Azurite Table service is successfully listening at http://0.0.0.0:10002
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:19 +0000] "GET /devstoreaccount1?comp=properties&restype=account HTTP/1.1" 200 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:19 +0000] "GET /devstoreaccount1?comp=properties&restype=account HTTP/1.1" 200 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:22 +0000] "GET /devstoreaccount1?comp=list&include=metadata HTTP/1.1" 200 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:22 +0000] "GET /devstoreaccount1?comp=list&include=metadata&timeout=30 HTTP/1.1" 200 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:22 +0000] "GET /devstoreaccount1/%24logs?restype=container HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:22 +0000] "GET /devstoreaccount1/%24blobchangefeed?restype=container HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:25 +0000] "GET /devstoreaccount1?comp=list&include=metadata HTTP/1.1" 200 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:25 +0000] "GET /devstoreaccount1/%24logs?restype=container HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:25 +0000] "GET /devstoreaccount1/%24blobchangefeed?restype=container HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:25 +0000] "GET /devstoreaccount1?restype=service&comp=properties HTTP/1.1" 200 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/Tables HTTP/1.1" 200 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsCapacityBlob()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsHourPrimaryTransactionsFile()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsHourPrimaryTransactionsBlob()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsHourPrimaryTransactionsQueue()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsHourSecondaryTransactionsBlob()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsHourPrimaryTransactionsTable()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsHourSecondaryTransactionsFile()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsHourSecondaryTransactionsQueue()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsMinutePrimaryTransactionsBlob()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsHourSecondaryTransactionsTable()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsMinutePrimaryTransactionsQueue()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsMinutePrimaryTransactionsFile()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsMinutePrimaryTransactionsTable()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsMinuteSecondaryTransactionsFile()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsMinuteSecondaryTransactionsQueue()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsMinuteSecondaryTransactionsBlob()?$top=1 HTTP/1.1" 404 -
azurite-internalapp  | 172.18.0.1 - - [18/Dec/2024:14:32:26 +0000] "GET /devstoreaccount1/%24MetricsMinuteSecondaryTransactionsTable()?$top=1 HTTP/1.1" 404 -
```

Selecting "queue" tree provides no logging, error message, etc. The timeout that is produced after ~15 minutes is pasted in section "Actual Experience"

**Comment by gallowasusc** [comment]
Did not think about images.

Where queue service gets stuck loading:

![Image](https://github.com/user-attachments/assets/3ca7ea0c-362b-4b04-89b0-c6120c69589c)

After about 5 minutes:

![Image](https://github.com/user-attachments/assets/7280c7e5-45bf-4a7a-a309-f72d0a823719)

Production resources have no issues connecting to queue services.

**Comment by craxal** [comment]
@gallowasusc Not able to reproduce on my end. I'm able to list and create queues as expected.

Click on the Docker node in the tree view and check the endpoints. Do they look right to you?

<img width="431" alt="Image" src="https://github.com/user-attachments/assets/43c3d8d5-b211-43bb-9e30-419d58cb3415" />

Also try creating a manual connection to the docker container to make sure that the queue service can receive requests.

**Comment by gallowasusc** [comment]
Your comment about manual connections inspired me to make the troubleshooting even more vanilla. I removed the docker container from the equation and attempted to spin up azurite on the host. The queue service would not start. I ran the following:

`sudo lsof -i -P | grep LISTEN | grep :$PORT`

and discovered a vscode process listening on port 10001. It persisted through restarts and an uninstall of the vscode extension. Ran a kill command on that process, retried the docker container and now queues work as intended.

![Image](https://github.com/user-attachments/assets/6456e0a2-fade-4fa3-bbc4-674e3d7065fa)

I'd been troubleshooting this one for days, but I'm glad it was just a user error. Thank you for your time!
